kernel/page_allocator/src/lib.rs

```diff
@@ -810,46 +1050,66 @@ pub fn allocate_pages_by_bytes_deferred(
 
 
 /// Allocates the given number of pages with no constraints on the starting virtual address.
-/// 
-/// See [`allocate_pages_deferred()`](fn.allocate_pages_deferred.html) for more details. 
+///
+/// See [`allocate_pages_deferred()`](fn.allocate_pages_deferred.html) for more details.
 pub fn allocate_pages(num_pages: usize) -> Option<AllocatedPages> {
 	allocate_pages_deferred(AllocationRequest::Any, num_pages)
 		.map(|(ap, _action)| ap)
 		.ok()
 }
 
+pub fn allocate_2mb_pages(num_pages: usize) -> Option<AllocatedPages> {
+	let num_pages = num_pages * 512;
+	if let Some(mut ap) = allocate_pages_deferred(AllocationRequest::Any, num_pages)
+		.map(|(ap, _action)| ap)
+		.ok() {
+			ap.to_2mb_allocated_pages();
+			return Some(ap);
+		}
+	None
+}
+
+pub fn allocate_1gb_pages(num_pages: usize) -> Option<AllocatedPages> {
+	let num_pages = num_pages * 512 * 512;
+	if let Some(mut ap) = allocate_pages_deferred(AllocationRequest::Any, num_pages)
+		.map(|(ap, _action)| ap)
+		.ok() {
+			ap.to_1gb_allocated_pages();
+			return Some(ap);
+		}
+	None
+}
```

Comment:
These functions aren't correct due to lack of checking alignment, and also aren't necessary. Kindly remove them. 

Instead, what we should do (for now, until we have a proper `AllocationRequest::AlignedAt` variant) is to implement these as a fallible conversion method on the `AllocatedPages` type. This is the easiest way, as all we'll need to do is:
1. call a regular `allocate_pages` function to obtain a sufficiently large and properly aligned `AllocatedPages` consisting of 4KiB normal pages
2. call the conversion method on the returned pages, which will check the alignment conditions for both the start and end page, and adjust the internal `PageRangeSized` object to be the correct variant.

To give you a more specific example, here's some pseudo-ish code for what i mean:
```rust
impl AllocatedPages {
    ...

    pub fn try_into_2mib_huge_pages(self) -> Result<Self, ErrorOfSomeKind> {
        match self.pages {
            PageRangeSized::Normal4KiB(range) => {
                if range.start_address().is_aligned_to(PAGE_SIZE * 512) 
                   && range.end().start_address().is_aligned_to(PAGE_SIZE * 512)
                {
                    Ok(AllocatedPages {
                        pages: PageRangeSized::Huge2MiB(PageRange::new(
                            Page::containing_address(range.start_address()),
                            Page::containing_address(range.end_address()), // need to add this function to PageRange, or you can calculate it here
                        ),
                    })
                }
            }
            PageRangeSized::Huge2MiB(range) => Ok(range),
            PageRangeSized::Huge1GiB(range) => {
                // todo: infallible conversion from 1G to 2M huge pages
                ...
            }
        }
    }
}
```